stacks-signer/src/v0/signer.rs

```diff
@@ -617,6 +667,85 @@ impl Signer {
         }
     }
 
+    /// Check the current tracked submitted block proposal to see if it has timed out.
+    /// Broadcasts a rejection and marks the block locally rejected if it has.
+    fn check_submitted_block_proposal(&mut self) {
+        let Some((block_proposal, block_submission)) = self.submitted_block_proposal.clone() else {
```

Comment:
Could you `take` here instead of cloning? Then in the case where it's not expired yet, just put it back.

Reply:
Done! :D